samples/cpp/tutorial_code/ImgTrans/HoughLines_Demo.cpp

```diff
@@ -131,3 +143,39 @@ void Probabilistic_Hough( int, void* )
 
    imshow( probabilistic_name, probabilistic_hough );
 }
+
+/**
+ * @function Weighted_Hough
+ * This can detect lines based on the edge intensities.
+ */
+void Weighted_Hough( int, void* )
+{
+  vector<Vec2f> s_lines;
+
+  /// prepare
+  Mat edge_img;
+  convertScaleAbs(sobel_edge, edge_img );
+  // use same threshold for edge with Hough.
+  threshold( edge_img, edge_img, e_trackbar, 255, cv::THRESH_TOZERO);
+  cvtColor( edge_img, weighted_hough, COLOR_GRAY2BGR );
+
+  /// 3. Use Weighted Hough Transform
+  const bool use_edgeval{true};
+  HoughLines( edge_img, s_lines, 1, CV_PI/180, min_threshold + w_trackbar, 0, 0,
+  	      0, CV_PI, use_edgeval);
+
+  /// Show the result
+  for( size_t i = 0; i < s_lines.size(); i++ )
+     {
+      float r = s_lines[i][0], t = s_lines[i][1];
+      double cos_t = cos(t), sin_t = sin(t);
+      double x0 = r*cos_t, y0 = r*sin_t;
+      double alpha = 1000;
+
+       Point pt1( cvRound(x0 + alpha*(-sin_t)), cvRound(y0 + alpha*cos_t) );
+       Point pt2( cvRound(x0 - alpha*(-sin_t)), cvRound(y0 - alpha*cos_t) );
+       line( weighted_hough, pt1, pt2, Scalar(255,0,0), 3, LINE_AA );
```

Comment:
Classic Hough transform and probablistic versions return lines in Cartesian coordinate system. You version implements polar version. I propose to return back to Cartesian and I'll merge the patch.

Reply:
I'm very sorry for not being able to correct nor respond.
When I have time, I will definitely fix based on your request.